### DISCUSSION TITLE
[API Request] Bulk set ratings

### DISCUSSION THREAD

**Discussion by jeffvli** [opensubsonic/open-subsonic-api#15] [API requests]
### Type of change

API tweak

### Proposal description

The `setRating` endpoint currently only supports setting a rating for one item at a time (not sure if some servers have already implemented this to support bulk rating). UX could be greatly improved by allowing users to bulk set ratings for library items.

I believe it can be improved to function similar to the `star` endpoint to set ratings by a batch of items.

### Backward compatibility

- [X] No backward compatibility impact.

### setRating
This will be updated to function similarly to the `star` endpoint which allows for multiple ids to be starred.

```
Parameters to add:
id - ID of song/folder or item to set rating for (allow multiple)
albumId -- ID of album to set rating for (allow multiple)
artistId -- ID of artist to set rating for (allow multiple)
```

Examples:
```
GET /setRating?albumId=1000&albumId=1001&albumId=1002
```

```
POST /setRating
{
  "albumId": ["1000", "1001", "1002"]
}

```

**Comment by dweymouth** [comment]
Speaking of ratings, we should support ratings on ArtistID3 and AlbumID3 too. Maybe some servers already support it but officially only tracks and MusicFolders can be rated. I can start a new discussion for this

**Comment by Tolriq** [comment]
As discussed in another thread, we should also define proper support for HTTP POST support before expanding too much the bulk commands. Some servers have really low query size limits and can quickly become problematic.
